modules/core/src/main/java/org/apache/ignite/internal/processors/cache/persistence/DataRegion.java

```diff
@@ -80,4 +87,25 @@ public DataRegionMetricsImpl memoryMetrics() {
     public PageEvictionTracker evictionTracker() {
         return evictionTracker;
     }
+
+    /**
+     * Increase the pool of empty pages, if it is too small.
+     *
+     * @return {@code true} if increased.
+     */
+    public synchronized boolean increaseEmptyPagesPool() {
+        if (cfg != null && pageMem != null
+            && emptyPagesPoolSize < (cfg.getMaxSize() / pageMem.systemPageSize() * (1 - cfg.getEvictionThreshold()))) {
```

Comment:
Why is this right border chosen `(cfg.getMaxSize() / pageMem.systemPageSize() * (1 - cfg.getEvictionThreshold()))`?
For instance, `IgniteCacheDatabaseSharedManager#checkRegionEvictionProperties` assumes that only ~10% of pages can be used (yes, if the default value of the eviction threshold is used, then the right border will be the same).
I think it should be aligned for both cases.